src/main/java/com/example/event_manager/service/BillingReportServiceImpl.java

```diff
@@ -56,4 +65,80 @@ public String generateXmlForBillingReportSchema(final BillingRaportSchema brs) {
       byteArrayOutputStream.close();
     }
   }
+
+  @Override
+  public List<BillingForm> filterBillingFormList(List<BillingForm> oldBillingForm,
+      BillingFilter billingFilter) {
+    List<BillingForm> filtered = oldBillingForm
+        .stream()
+        .filter(bf -> {
+          if (billingFilter.getBillingType().equals("all")) {
```

Comment:
more secure approach is to put contant value at left side during comparison, like:
"all".equals(billingFilter.getBillingType())
you can avoid NPE then